lib/core/application/event/claim_rewards_bloc/claim_rewards_bloc.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:app/core/domain/event/entities/event_reward_use.dart';
+import 'package:app/core/domain/event/repository/event_reward_repository.dart';
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/domain/user/user_repository.dart';
+import 'package:app/graphql/backend/schema.graphql.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'claim_rewards_bloc.freezed.dart';
+
+class ClaimRewardsBloc extends Bloc<ClaimRewardsEvent, ClaimRewardsState> {
```

Comment:
This bloc obviously use for getting rewards, should we rename it ? "Claim" is another action